src/container/containerManager.ts

```diff
@@ -128,8 +137,20 @@ export class ContainerManager {
         }
     }
 
-    private constructBuildCmd(dockerfilePath: string, imageName: string, contextDir: string): string {
-        return `docker build --rm -f \"${Utility.adjustFilePath(dockerfilePath)}\" -t ${imageName} \"${Utility.adjustFilePath(contextDir)}\"`;
+    private constructBuildCmd(dockerfilePath: string, imageName: string, contextDir: string, extraOptions ?: string[]): string {
+        let optionString: string = "";
+        if (extraOptions !== undefined) {
+            const filteredOption = extraOptions.filter((value, index) => {
+                const trimmed = value.trim();
+                return ! (trimmed.startsWith("--rm") ||
+                            trimmed.startsWith("--tag") ||
+                            trimmed.startsWith("-t") ||
+                            trimmed.startsWith("--file") ||
+                            trimmed.startsWith("-f"));
```

Comment:
This will also filter options like `--target` and `--force-rm`, right?

Reply:
Sorry. I am wrong. These start with double `-`

Reply:
But I still believe we should better check the boundary of the options. Say, in the future, a new option `--file-balabala` is added, it will be filtered.

Reply:
Agree, We could add a space (e.g. `trimmed.startsWith("--file ")`) to check

Reply:
I guess simply adding a space may not be enough, because the option could have default values. For example, `--rm` has the same effect as `--rm true`.